### PR TITLE
Add Pathak-Wagner Regularizer to ACForce

### PR DESCRIPTION
**Pull request by rcclay**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

This adds the Pathak-Wagner regularizer to all ACForce components, thus allowing the user to mitigate the formal infinite variance problems of this estimator.  Access to this feature is done by specifying `epsilon=[Real > 0]` in the ACForce XML section.  

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature
- Testing changes (e.g. new unit/integration/performance tests)
- Documentation changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Intel Xeon, RHEL7, Real and Complex builds.  

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCHamiltonians/ACForce.cpp [hunk 1]
```diff
@@ -63,9 +68,12 @@ bool ACForce::put(xmlNodePtr cur)
   attr.add(useSpaceWarp_, "spacewarp", {false}); //"yes" or "no"
   attr.add(swpow, "swpow");                      //Real number"
   attr.add(delta_, "delta");                     //Real number"
+  attr.add(reg_epsilon_,"epsilon");
   attr.add(fastDerivatives_, "fast_derivatives", {false});
   attr.put(cur);
-
+  if( reg_epsilon_ < 0)
+    throw std::runtime_error("ACForce::put(): epsilon<0 not allowed.")
+;
```
Comment (ye-luo) [inline src/QMCHamiltonians/ACForce.cpp:76]: Semicolon on a separate line?
Please clang-format all the touched files. I saw a few outliers.

File: src/QMCHamiltonians/ACForce.cpp [hunk 2]
```diff
@@ -168,12 +186,46 @@ void ACForce::setParticlePropertyList(PropertySetType& plist, int offset)
...
+
+  RealType regvalue=0.0;
+  //x = grad(logpsi)/|grad(logpsi)|^2 = 1/|grad(logpsi)|.
+  //
+  //Argument of polynomial is x/epsilon=1/(epsilon*|grad(logpsi)|)
+  double xovereps=1.0/(epsilon*gmag);
+  if (xovereps >= 1.0)
+  {
+    regvalue = 1.0; 
+  }
```
Comment (ye-luo) [inline src/QMCHamiltonians/ACForce.cpp:222]: Remove `{}` around one line.

File: src/QMCHamiltonians/ACForce.h [hunk 3]
```diff
@@ -62,13 +62,15 @@ class ACForce : public OperatorBase
  *  isn't sufficient.  We override it here. **/
   void add2Hamiltonian(ParticleSet& qp, TrialWaveFunction& psi, QMCHamiltonian& targetH) final;
 
+  RealType compute_regularizer_f(const ParticleGradient & G, const RealType epsilon);
```
Comment (ye-luo) [inline src/QMCHamiltonians/ACForce.h:65]: Please document this function. Can this function marked static?

File: src/QMCHamiltonians/ACForce.cpp (hunk 1 above)
Comment (rcclay) [reply src/QMCHamiltonians/ACForce.cpp:76]: Oops.  I clang-formatted, but forgot to push up the commit.

File: src/QMCHamiltonians/ACForce.h (hunk 3 above)
Comment (rcclay) [reply src/QMCHamiltonians/ACForce.h:65]: For sure.  I'll document and make static.

File: src/QMCHamiltonians/ACForce.cpp (hunk 2 above)
Comment (rcclay) [reply src/QMCHamiltonians/ACForce.cpp:222]: Done

File: src/QMCHamiltonians/ACForce.h (hunk 3 above)
Comment (rcclay) [reply src/QMCHamiltonians/ACForce.h]: Done

File: src/QMCHamiltonians/ACForce.cpp (hunk 1 above)
Comment (rcclay) [reply src/QMCHamiltonians/ACForce.cpp]: Done.

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

File: src/QMCHamiltonians/ACForce.cpp [hunk 4]
```diff
@@ -168,12 +185,44 @@ void ACForce::setParticlePropertyList(PropertySetType& plist, int offset)
...
+  RealType regvalue = 0.0;
+  //x = grad(logpsi)/|grad(logpsi)|^2 = 1/|grad(logpsi)|.
+  //
+  //Argument of polynomial is x/epsilon=1/(epsilon*|grad(logpsi)|)
+  double xovereps = 1.0 / (epsilon * gmag);
+  if (xovereps >= 1.0)
+    regvalue = 1.0;
+  else
+  {
+    //There's a discrepancy between AIP Advances 10, 085213 (2020) and arXiv:2002.01434 for polynomial.
```
Comment (prckent) [inline src/QMCHamiltonians/ACForce.cpp:222]: Interesting

**Pull request merged**